docs/adr/adr_0011.md

```diff
@@ -0,0 +1,167 @@
+<!--
+SPDX-FileCopyrightText: 2021 iteratec GmbH
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+# ADR-0011: Version Numbers
+
+| <!-- -->       | <!-- --> |
+|----------------|----------|
+| **Status**:    | DRAFT |
+| **Date**:      | 2022-01-18 |
+| **Author(s)**: | Max Maass <max.maass@iteratec.com> |
+
+## Context
+Software version numbers should serve as an indicator of compatibility between different versions, both during operation and while updating to a newer release.
+At the moment, the _secureCodeBox_ is following the [Semantic Versioning (semver)][semver] approach.
+According to semver, version numbers are of the format MAJOR.MINOR.PATCH, and the different places are incremented as follows:
+
+> 1. MAJOR version when you make incompatible API changes,
+> 2. MINOR version when you add functionality in a backwards compatible manner, and
+> 3. PATCH version when you make backwards compatible bug fixes.
+
+However, the architecture of _secureCodeBox_, with its operator and many individual scanners that can be installed and used separately, makes this seemingly simple versioning system more difficult.
+For example, we need to answer the following questions:
+
+1. Are breaking changes in the parameterization of a scanner (e.g. nmap, Gitleaks) a breaking change for the entire _secureCodeBox_ project?
+2. Are changes to the output format of a single scanner a breaking change for the entire project?
+3. In an environment where operator and scanners aren't necessarily using the same version number because the operator is controlled by a different team than the scanners (a setup that we want to support), how do we indicate compatibility between different versions of operator and scanner?
+
+Depending on how these questions are answered, different versioning schemes are possible.
+
+### Option 1: SemVer With Major Version Indicating Overall Compatibility Of All Components
+The basic premise for this versioning scheme would be:
+
+> Any change that requires manual actions from at least one user of the SCB to keep existing workflows running after an update is considered a breaking change and requires a MAJOR release.
+
+This manual action may include making changes to scan definitions, or to systems that are ingesting data from the SCB findings of a scanner.
+It sees the entire _secureCodeBox_ as **one large piece of software with many components** that are all equally important to the overall compatibility, and where all components are (usually) updated in lockstep.
+As illustration, here are a few examples and what kind of release they would require:
+
+| Action                                                                             |Version |
+|------------------------------------------------------------------------------------|--------|
+| A scanner changes how it is parameterized                                          | Major  |
+| A scanner changes what data it returns                                             | Major  |
```

Comment:
Is this really major because this is abstracted by our parser layer. So in an ideal world we can compensate this change inside the parser.

Reply:
This was referring more to a case like gitleaks, where data is completely removed from the output, or the format is changed significantly (i.e., something that cannot be compensated through changes in the parser). I'll reword this part of the table.

Reply:
Is the new version clearer?